tests/testthat/test-cache.R

```diff
@@ -53,116 +53,93 @@ test_that("clear_cache deletes the directory if no file is given", {
 # `load_cache`
 
 test_that("load_cache loads the saved file in a new empty environment", {
-  with_mock(
-    `lintr::read_settings` = function(...) invisible(...),
-
-    e1 <- new.env(parent = emptyenv()),
-    assign("x", "foobar", envir = e1),
-    d1 <- tempfile(pattern = "lintr_cache_"),
-    f1 <- "R/test.R",
-    save_cache(cache = e1, file = f1, path = d1),
-    e2 <- load_cache(file = f1, path = d1),
-
-    expect_equal(ls(e2), "x"),
-    expect_equal(e2[["x"]], "foobar")
-  )
+  e1 <- new.env(parent = emptyenv())
+  e1[["x"]] <- "foobar"
+  d1 <- tempfile(pattern = "lintr_cache_")
+  f1 <- "R/test.R"
+  save_cache(cache = e1, file = f1, path = d1)
+  e2 <- load_cache(file = f1, path = d1)
+
+  expect_environments_equal(e2, e1)
```

Comment:
Why not stick with `expect_equal(as.list(e1), as.list(e2))` and add a `# TODO` once r-lib/waldo/pull/133 makes it to CRAN?

Reply:
sure, done